src/app/chip_data_model.gni

```diff
@@ -390,6 +390,13 @@ template("chip_data_model") {
           "${_app_root}/clusters/${cluster}/thread-network-diagnostics-provider.cpp",
           "${_app_root}/clusters/${cluster}/thread-network-diagnostics-provider.h",
         ]
+      } else if (cluster == "thread-border-router-management-server") {
+        sources += [
+          "${_app_root}/clusters/${cluster}/${cluster}.cpp",
+          "${_app_root}/clusters/${cluster}/${cluster}.h",
```

Comment:
Please list thread-border-router-managemenet-server.h/.cpp explicitly, and don't reuse this `${cluster}.cpp/.h` as it makes it harder to find which target depends on those files.